devito/propagator.py

```diff
@@ -34,6 +34,7 @@ class Propagator(object):
     def __init__(self, name, nt, shape, spc_border=0, time_order=0,
                  time_dim=None, space_dims=None, forward=True, compiler=None,
                  profile=False, cache_blocking=False, block_size=5):
+        self.dtype = np.float32  # set the default precision to single
```

Comment:
No, Devito provides two API layers, so defaults are required at both levels:
* The high-level via the Operator is purely symbolic, permitting data objects accessed as functions, and thus enabling all the SymPy magic.
* The Propagator API is a bit more restrictive, since it requires data to be accessed as `Indexed` objects and is more stringent with respect to time indexing, etc. It does however allow more low-level operations that are otherwise hard to capture in pure symbolic sympy expressions.